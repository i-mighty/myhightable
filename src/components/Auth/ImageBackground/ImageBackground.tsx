import styled from 'styled-components/native';
import {
  ImageBackground as RNImageBackground,
  ImageSourcePropType,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import colorOpacity from '@src/utils/color-opacity';
import { FC } from 'react';
import { Content } from 'native-base';
import React from 'react';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import ActionDescription from '../ActionDescription/ActionDescription';

const StyledImageBackground = styled(RNImageBackground)`
  height: 100%;
  width: 100%;
`;

const StyledContent = styled(Content)`
  height: 100%;
  width: 100%;
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
`;

interface Props {
  source: ImageSourcePropType;
  animation?: Animatable.Animation;
  authState?: string;
  testID?: string;
}

const AnimatedImageBackground = Animatable.createAnimatableComponent(
  StyledImageBackground,
);

const ImageBackground: FC<Props> = ({
  source,
  children,
  animation,
  testID,
}) => {
  return (
    <AnimatedImageBackground
      source={source}
      animation={animation || 'fadeIn'}
      testID={testID}>
      <Header />
      <StyledContent
        contentContainerStyle={{
          flex: 1,
        }}>
        {children}
      </StyledContent>
    </AnimatedImageBackground>
  );
};

export default ImageBackground;
