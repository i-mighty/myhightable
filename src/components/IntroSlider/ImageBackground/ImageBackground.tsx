import styled from 'styled-components/native';
import {
  ImageBackground as RNImageBackground,
  ImageSourcePropType,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import colorOpacity from '@src/utils/color-opacity';
import { FC } from 'react';
import { Content } from 'native-base';
import React from 'react';
import Header from '../Header/Header';

const StyledImageBackground = styled(RNImageBackground)`
  height: 100%;
  width: 100%;
`;

const StyledContent = styled(Content)`
  height: 100%;
  width: 100%;
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
`;

interface Props {
  source: ImageSourcePropType;
  animation?: Animatable.Animation;
  toSkip?: boolean;
  skipAction?: () => void;
  testID?: string;
}

const AnimatedImageBackground = Animatable.createAnimatableComponent(
  StyledImageBackground,
);

const ImageBackground: FC<Props> = ({
  source,
  children,
  animation,
  skipAction,
  toSkip,
  testID,
}) => {
  return (
    <AnimatedImageBackground
      source={source}
      animation={animation || 'fadeIn'}
      testID={testID}>
      {toSkip && <Header buttonText="Skip" onButtonClick={skipAction} />}
      <StyledContent
        contentContainerStyle={{
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        {children}
      </StyledContent>
    </AnimatedImageBackground>
  );
};

export default ImageBackground;
