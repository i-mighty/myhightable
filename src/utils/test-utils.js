import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
