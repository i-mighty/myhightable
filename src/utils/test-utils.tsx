import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
