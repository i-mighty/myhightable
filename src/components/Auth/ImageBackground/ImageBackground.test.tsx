import ImageBackground from './ImageBackground';
import { renderWithProps } from '@src/utils/test-utils';

const defaultProps = {
  source: require('@src/assets/Intro/cafe.jpg'),
  testID: 'imageBackground',
};

describe('ImageBackground', () => {
  it('renders properly', () => {
    const { getByTestId, baseElement } = renderWithProps(
      ImageBackground,
      defaultProps,
    );
    expect(getByTestId(defaultProps.testID)).toBeDefined();
    expect(getByTestId(defaultProps.testID)).toHaveProp(
      'source',
      defaultProps.source,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('renders with Banner Header', () => {
    const { getByTestId, baseElement } = renderWithProps(
      ImageBackground,
      defaultProps,
    );
    expect(getByTestId(defaultProps.testID)).toBeDefined();
    expect(getByTestId('banner')).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });
});
