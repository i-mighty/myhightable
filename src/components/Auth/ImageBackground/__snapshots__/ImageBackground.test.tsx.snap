// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBackground renders properly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Image
        animation="fadeIn"
        source={
          Object {
            "testUri": "../../../src/assets/Intro/cafe.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="imageBackground"
      />
      <View>
        <SafeAreaView
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.5)",
            }
          }
        >
          <View
            noLeft={true}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingLeft": 6,
                  "paddingRight": 10,
                  "paddingTop": 18,
                  "right": 0,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "width": "100%",
                },
              ]
            }
            transparent={true}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <SvgMock
                height={38}
                testID="banner"
                width={149}
              />
            </View>
          </View>
        </SafeAreaView>
      </View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "flex": 1,
          }
        }
      >
        <ScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableOnAndroid={false}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          showsVerticalScrollIndicator={true}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          viewIsInsideTabBar={false}
        />
      </SafeAreaView>
    </View>
  </View>
</View>
`;

exports[`ImageBackground renders with Banner Header 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Image
        animation="fadeIn"
        source={
          Object {
            "testUri": "../../../src/assets/Intro/cafe.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="imageBackground"
      />
      <View>
        <SafeAreaView
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.5)",
            }
          }
        >
          <View
            noLeft={true}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingLeft": 6,
                  "paddingRight": 10,
                  "paddingTop": 18,
                  "right": 0,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "width": "100%",
                },
              ]
            }
            transparent={true}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <SvgMock
                height={38}
                testID="banner"
                width={149}
              />
            </View>
          </View>
        </SafeAreaView>
      </View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "flex": 1,
          }
        }
      >
        <ScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableOnAndroid={false}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          showsVerticalScrollIndicator={true}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          viewIsInsideTabBar={false}
        />
      </SafeAreaView>
    </View>
  </View>
</View>
`;
